core/src/driver/stablex_driver.rs

```diff
@@ -127,16 +154,32 @@ impl<'a> StableXDriverImpl<'a> {
         };
 
         let submitted = if let Some(objective_value) = verified {
+            if min_solution_submit_time > Duration::from_secs(0) {
+                if let Ok(duration) = (batch_to_solve.solve_start_time() + min_solution_submit_time)
```

Comment:
Is there a reason for the nested if? Could we just have the `if let Ok()` statement?

Reply:
I added a comment to explain this:

            // In the e2e test batch ids don't always correspond to the correct batch id based on
            // real time (we fake advancement of time) so without the outer if we would be waiting
            // a long time. There will likely be a refactor that lets the scheduler decider when
            // solutions should be submitted which fixes this problem in a better way.